### ISSUE TITLE
MySQL doesn't support "CREATE INDEX IF NOT EXISTS"

### ISSUE THREAD

**Issue by niclash** [jOOQ/jOOQ#12387]
### Expected behavior
With MySQL dialect/server, `createIndexIfNotExists(name)...` should create an index.
Tested on MySQL 5.7.34, but according to MySQL documentation version 8.0 still doesn't support it.

### Actual behavior
Exception
```
org.jooq.exception.DataAccessException: SQL [create index if not exists `IDX_FlatEntity_ASSOCS` on `FlatEntity_ASSOCS`(`_identity`)]; You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'if not exists `IDX_FlatEntity_ASSOCS` on `FlatEntity_ASSOCS`(`_identity`)' at line 1
	at org.jooq_3.14.14.MYSQL.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2903)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:757)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:389)
```

### Versions

- jOOQ: 3.14.14  (3.15.x on Maven Central has Java 11 class files in it)
- Java: 8
- Database (include vendor): MySQL 5.7.34
- OS: Debian 11
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks. We can't emulate these kinds of statements on MySQL yet. They would depend on the availability of an execution model configuration (see https://github.com/jOOQ/jOOQ/issues/9645). As MySQL doesn't support anonymous blocks, we'd have to create a procedure, run the procedural logic that prevents failure, and drop the procedure again. Alternatively, we could run several statements.

Note we didn't document that the statement was supported. Check out https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#createIndexIfNotExists()

The annotation there says:

```java
@Support({AURORA_POSTGRES,COCKROACHDB,DB2,FIREBIRD,H2,HANA,HSQLDB,IGNITE,INFORMIX,MARIADB,ORACLE,POSTGRES,SQLDATAWAREHOUSE,SQLITE,SQLSERVER,SYBASE})
```

So, this isn't a bug.

> * jOOQ: 3.14.14  (3.15.x on Maven Central has Java 11 class files in it)

Starting from jOOQ 3.15, our baseline for the jOOQ Open Source Edition is Java 11. For continued Java 8 usage, please use the jOOQ Professional Edition from here: https://www.jooq.org/download/versions.

The commercial distributions also continue supporting MySQL 5.7, see https://www.jooq.org/download/support-matrix

**Comment by lukaseder** [comment]
An emulation for this statement has been implemented in jOOQ 3.19:

- https://github.com/jOOQ/jOOQ/issues/11251

The emulation depends on commercial only features around procedural logic generation, as we have to create a procedure that catches any exceptions, call the procedure, and drop it again, since MySQL doesn't support anonymous blocks yet.

Closing this as a duplicate of #11251
